pkg/job/job_manager.go

```diff
@@ -0,0 +1,112 @@
+package job
+
+import (
+	"os/exec"
+	"sync"
+
+	"github.com/MinhNghiaD/jobworker/pkg/log"
+	"github.com/google/uuid"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/genproto/googleapis/rpc/errdetails"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// JobsManager is the interface for the externals to create and access to job
+type JobsManager interface {
+	// Create a new job and Start running it in the background with the command and its arguments, then return its ID
+	CreateJob(command string, args []string, owner string) (string, error)
+	// Return a created job corresponding to the ID, if the job is not existed, return nil
+	GetJob(ID string) (Job, bool)
+	// Perform clean up
+	Cleanup() error
+}
+
+// JobsManagerImpl is the implementation of the Jobs manager
+type JobsManagerImpl struct {
+	jobStore    map[string]Job
+	logsManager *log.LogsManager
+	mutex       sync.RWMutex
+}
+
+// NewManager creates a new Job Manager with its dedicated space
+func NewManager() (JobsManager, error) {
+	logsManager, err := log.NewManager()
+	if err != nil {
+		return nil, err
+	}
+
+	return &JobsManagerImpl{
+		jobStore:    make(map[string]Job),
+		logsManager: logsManager,
+	}, nil
+}
+
+// CreateJob starts a new job in the background with the command and its arguments, then associates the job with its owner
+func (manager *JobsManagerImpl) CreateJob(command string, args []string, owner string) (string, error) {
+	if manager.logsManager == nil {
+		return "", status.Errorf(codes.Unavailable, "Log Manager is not iniatiated")
```

Comment:
nit: this breaks the abstraction a bit - gRPC error codes should live at gRPC service layer